src/main/java/net/neoforged/neoforge/server/command/TPSCommand.java

```diff
@@ -5,59 +5,94 @@
+        var src = context.getSource();
+        src.sendSuccess(() -> createComponent(src.getServer(), dimension), false);
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static Component createComponent(MinecraftServer server, @Nullable ServerLevel dimension) {
+        long[] times;
+        TickRateManager tickRateManager;
+
+        if (dimension == null) {
+            times = server.getTickTimesNanos();
+            tickRateManager = server.tickRateManager();
+        } else {
+            var dimensionTimes = server.getTickTime(dimension.dimension());
+            times = dimensionTimes == null ? UNLOADED : dimensionTimes;
+            tickRateManager = dimension.tickRateManager();
+        }
+
+        var tickTime = Stats.meanOf(times) / TimeUtil.NANOSECONDS_PER_MILLISECOND;
+        var tps = TimeUtil.MILLISECONDS_PER_SECOND / Math.max(tickTime, tickRateManager.millisecondsPerTick());
+        var tickTimeComponent = Component.literal(TIME_FORMATTER.format(tickTime)).withColor(CommonColors.LIGHT_GRAY);
+        var tpsComponent = Component.literal(TIME_FORMATTER.format(tps)).withColor(calculateTPSColor(tickRateManager, tps));
+
+        MutableComponent component;
+
+        if (dimension == null)
+            component = Component.translatable("commands.neoforge.tps.overall", tpsComponent, tickTimeComponent);
+        else {
+            var dimensionType = dimension.dimensionTypeRegistration();
 
-        if (times == null) // Null means the world is unloaded. Not invalid. That's taken care of by DimensionArgument itself.
-            times = UNLOADED;
+            var dimensionName = Component.empty().append(dimension.getDescription()).withStyle(style -> style
+                    .withColor(ChatFormatting.GREEN)
+                    .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.translatable(
+                            "commands.neoforge.tps.dimension.tooltip",
+                            dimension.dimension().location().toString(),
+                            dimensionType.getRegisteredName()))));
 
-        final Registry<DimensionType> reg = cs.registryAccess().registryOrThrow(Registries.DIMENSION_TYPE);
-        double worldTickTime = mean(times) * 1.0E-6D;
-        double worldTPS = TimeUtil.MILLISECONDS_PER_SECOND / Math.max(worldTickTime, dim.tickRateManager().millisecondsPerTick());
-        cs.sendSuccess(() -> Component.translatable("commands.neoforge.tps.summary.named", dim.getDescription(), reg.getKey(dim.dimensionType()).toString(), TIME_FORMATTER.format(worldTickTime), TIME_FORMATTER.format(worldTPS)), false);
+            component = Component.translatable("commands.neoforge.tps.dimension", dimensionName, tpsComponent, tickTimeComponent);
+        }
 
-        return 1;
+        return component.withStyle(style -> style.withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.translatable("commands.neoforge.tps.tooltip", tickRateManager.tickrate()))));
```

Comment:
Did you make sure that this hover event on the entire component does not prevent the dimension name hover event from showing up?

Reply:
Yes, you can still hover the dimension name to view the registry id and dimension type.
_See the updated "after" gif_